File: 프로그래머스/lv0/120905. n의 배수 고르기/n의 배수 고르기.js
function solution(n, numlist) {
    let result = numlist.filter( (element) => { return element%n === 0 });
    return result;
}
/*
    접근법 :
    .filter() : 콜백 함수를 통과해 true인 요소만 모아 새로운 배열 반환

    element(요소)%n === 0 즉, n으로 나누었을 때, 0인 요소 = n 의 배수이다.
*/

File: 프로그래머스/lv0/120905. n의 배수 고르기/n의 배수 고르기.java
import java.util.ArrayList;
class Solution {
    public ArrayList<Integer> solution(int n, int[] numlist) {
        ArrayList<Integer> list = new ArrayList();
        for(int i=0; i<numlist.length; i++) {
            if(numlist[i]%n == 0) {
                list.add(numlist[i]);
            }
        }
        return list;
    }
}
/*
    접근법 :
    n의 배수만 넣을 배열이 필요하다 But 몇개나 넣을지는 미정 -> ArrayList를 사용하자
*/